### ISSUE TITLE
Error configuring kubernetes auth for vault: No host provided

### ISSUE THREAD

**Issue by gowthamsubbu** [banzaicloud/bank-vaults#152]
@bonifaido We tried to initialize kubernetes auth method, with the below config which used to work and successfully configure kubernetes auth method.

Recently, we found the error in configurer pod logs which says 

level=error msg="error configuring vault: error configuring kubernetes auth for vault: error putting map[] kubernetes config into vault: Error making API request.\n\nURL: PUT https://vault.secrets:8200/v1/auth/kubernetes/config\nCode: 400. Errors:\n\n* no host provided"

As per the below line, we can find the KUBERNETES_SERVICE_HOST in the configurer pod. Also, we can see the hostname configured properly under kubernetes auth section in Vault.

https://github.com/gowthamsubbu/bank-vaults/blob/a3779e94a4133df48a12ecc3c28f326b392e6787/pkg/vault/vault.go#L447

![image](https://user-images.githubusercontent.com/34745306/45313142-0006bd00-b4fc-11e8-845a-2a2d6ad83bab.png)

Vault: 10.4
Bank-Vaults: 0.2.20

```yaml
auth:
    - type: kubernetes
      roles:
        # Allow every pod in the secrets namespace to use the secret kv store
        - name: Kube-Vault-ServiceAccount-Role
          bound_service_account_names: vault
          bound_service_account_namespaces: secrets
          policies: admin
          ttl: 1h
```

**Comment by bonifaido** [comment]
Hi @gowthamsubbu, the fix has been pushed in #153, after review I will merge it.

**Comment by bonifaido** [comment]
You're welcome, no problem!
